### PR TITLE
WES QC Workflow for AnVIL

### PR DESCRIPTION
**Pull request by jasonwalker80**
Added support for WES QC and generalized the whole workflow/repo to show support for either WGS or WES with a generic `is_wgs` input (boolean).

### CODE REVIEW

File: SingleSampleQc.wdl [hunk 1]
```diff
@@ -24,11 +24,12 @@ version 1.0
 ## - [VerifyBamID2](https://github.com/Griffan/VerifyBamID)
 
 # Git URL import
-import "https://raw.githubusercontent.com/genome/qc-analysis-pipeline/master/tasks/Qc.wdl" as QC
+#import "https://raw.githubusercontent.com/genome/qc-analysis-pipeline/master/tasks/Qc.wdl" as QC
+import "https://raw.githubusercontent.com/jasonwalker80/qc-analysis-pipeline/wes/tasks/Qc.wdl" as QC
```
Comment (tmooney) [inline SingleSampleQc.wdl:28]: It's a little bit strange that this change makes sense on the branch but not in the PR!  Does WDL support relative imports like CWL does?

File: SingleSampleQc.wdl [hunk 2]
```diff
@@ -2,10 +2,10 @@ version 1.0
 
 ## Copyright Broad Institute, 2018
```
Comment (tmooney) [inline SingleSampleQc.wdl:3]: Maybe this should say `Portions Copyright Broad Institute` at this point!

File: SingleSampleQc.wdl [hunk 3]
```diff
@@ -100,20 +102,37 @@ workflow WholeGenomeSingleSampleQc {
...
+    # QC the sample WES metrics since this IS NOT WGS
+    call QC.CollectHsMetrics as CollectHsMetrics {
+      input:
+        input_bam = BuildBamIndex.bam,
+        input_bam_index = BuildBamIndex.bam_index,
+        ref_fasta = ref_fasta,
+        ref_fasta_index = ref_fasta_index,
+        metrics_filename = base_name + ".hs_metrics",
+        target_interval_list = coverage_interval_list,
+        bait_interval_list = coverage_interval_list,
```
Comment (tmooney) [inline SingleSampleQc.wdl:131]: Is it okay using the same file for both of these?

File: SingleSampleQc.wdl [hunk 4]
```diff
@@ -100,20 +102,37 @@ workflow WholeGenomeSingleSampleQc {
...
+        ref_fasta_index = ref_fasta_index,
+        wgs_coverage_interval_list = coverage_interval_list,
+        read_length = read_length,
+        preemptible_tries = preemptible_tries
+    }
   }
-
+  
+  if (!is_wgs) {
+    # QC the sample WES metrics since this IS NOT WGS
```
Comment (tmooney) [inline SingleSampleQc.wdl:122]: Are these necessarily `WES` metrics?

File: SingleSampleQc.wdl (hunk 1 above)
Comment (acoffman) [reply SingleSampleQc.wdl:28]: WDL supports relative imports but most of the cloud platforms themselves don't.

Comment (tmooney) [reply SingleSampleQc.wdl:28]: That's a bummer; I guess for the present purpose the PR should be from a branch without 04de35a2f72f65d8dc3603360e0d657fba68530d whereas testing would want to include it.

File: SingleSampleQc.wdl (hunk 3 above)
Comment (jasonwalker80) [reply SingleSampleQc.wdl:131]: in testing this worked.

File: SingleSampleQc.wdl (hunk 2 above)
Comment (jasonwalker80) [reply SingleSampleQc.wdl]: fixed

File: SingleSampleQc.wdl (hunk 4 above)
Comment (jasonwalker80) [reply SingleSampleQc.wdl]: updated.

File: SingleSampleQc.wdl (hunk 1 above)
Comment (jasonwalker80) [reply SingleSampleQc.wdl]: now pointing back to master branch in preparation for the merge.

File: SingleSampleQc.wdl (hunk 3 above)
Comment (tmooney) [reply SingleSampleQc.wdl:129]: Makes sense that it would "work" in that it runs.  Mostly thinking it might make some of the results less meaningful.

**Pull request merged**